js/util/util.js

```diff
@@ -474,3 +475,27 @@ exports.warnOnce = function(message) {
         warnOnceHistory[message] = true;
     }
 };
+
+/**
+ * Calculates centroid of all the points
+ * @param {Array} points
+ * @returns {Array} [x,y] Centroid of all the points
+ * @private
+ */
+exports.centroid = function(points) {
+    assert(points !== null);
+    assert(points.length > 0);
+
+    if (points.length === 1) {
+        return [points[0].x, points[0].y];
+    }
+    // This tries to prevent overflow/underflow of the float
+    var xC = points[0].x, yC = points[0].y;
+    var xSum = 0, ySum = 0, len = 0;
+    points.forEach(function(coord) {
+        xSum += coord.x - xC;
+        ySum += coord.y - yC;
+        len++;
+    });
```

Comment:
Excellent thanks for pointing it out.  There was an error in the function on wikipedia that has now been corrected.  I have the additional changes but it causes tests to fail if we keep this change as symbol-placment: 'point'.  

The side effect is that lineStrings will use the centroid which I believe is incorrect.  I did ask if a symbol-placment: 'centroid' should be used in #1231.  This would require changes to the style spec as well, I may need some help/pointers to understand how that is also updated/released.